### DISCUSSION TITLE
anything that calls a class in java/sql is not being found

### DISCUSSION THREAD

**Discussion by PerryCameron** [wiverson/jtoolprovider-plugin#16] [Q&A] [answered]
It appears that the module java/sql is not being included.

The first thing I noticed was:
java.lang.ClassNotFoundException: java.sql.SQLException

I figured that java/sql wasn't being included, but was not completely sure, so I just changed the calls to Exception to get around it.
Then I called java/sql/Driver...and...
java.lang.ClassNotFoundException: java.sql.Driver

I see that in the App folder, in Contents/runtime/Contents/Home/lib looks to be the executable JVM dynamically linked libraries. I have no idea which one of those would contain the java/sql package to verify that it is missing.

**Comment by wiverson** [comment, marked as answer]
You will need to add java.sql to the bundled JVM.

https://github.com/wiverson/maven-jpackage-template/blob/main/pom.xml#L31

By default the trimmed down JVM doesn't include the SQL module (or a number of other modules). Here's a list of modules:

https://docs.oracle.com/en/java/javase/17/docs/api/index.html
